### PR TITLE
Add tags support

### PR DESCRIPTION
**Pull request by nopcoder**

### CODE REVIEW

File: graveler/pgrefs.go [hunk 1]
```diff
@@ -169,6 +168,58 @@ func (m *PGRefManager) ListBranches(ctx context.Context, repositoryID Repository
...
+	return commitID.(*CommitID), nil
+}
+
+func (m *PGRefManager) SetTag(ctx context.Context, repositoryID RepositoryID, tagID TagID, commitID CommitID) error {
+	_, err := m.db.Transact(func(tx db.Tx) (interface{}, error) {
+		_, err := tx.Exec(`
+			INSERT INTO graveler_tags (repository_id, id, commit_id)
+			VALUES ($1, $2, $3)
+				ON CONFLICT (repository_id, id)
+				DO UPDATE SET commit_id = $3`,
```
Comment (ozkatz) [inline graveler/pgrefs.go:196]: In git, tags are immutable. I think they can be force deleted,  but not updated

File: graveler/refresolver.go [hunk 2]
```diff
@@ -31,6 +32,14 @@ func (r reference) CommitID() CommitID {
 	return *r.commitID
 }
 
+type resolveHelperFunc func(context.Context, RefStore, RepositoryID, string) (Reference, error)
```
Comment (ozkatz) [inline graveler/refresolver.go:35]: I like the idea, not sure about the name though. Maybe `TypeResolverFunc`? Or something else that is descriptive

**Review by ozkatz — CHANGES_REQUESTED** [review]
That was fast! My only concern is tag immutability which I think we should maintain. looks great otherwise

File: graveler/pgrefs.go (hunk 1 above)
Comment (nopcoder) [reply graveler/pgrefs.go]: Moving SetTag to be CreateTag - will require delete and create for force update.

File: graveler/refresolver.go (hunk 2 above)
Comment (nopcoder) [reply graveler/refresolver.go]: refactored with a new name, please re-review this function

**Review by guy-har — APPROVED** [review]
LGTM

File: graveler/pgrefs_test.go [hunk 3]
```diff
@@ -250,6 +252,139 @@ func TestPGRefManager_ListBranches(t *testing.T) {
...
+		DefaultBranchID:  "master",
+	}, graveler.Branch{
+		CommitID: "c1",
+	}))
+
+	testutil.Must(t, r.CreateTag(ctx, "repo1", "v1", "c2"))
+
+	testutil.Must(t, r.DeleteTag(ctx, "repo1", "v1"))
+
+	commitID, err := r.GetBranch(ctx, "repo1", "v1")
```
[the remark this answers is not in the document]
Comment (nopcoder) [reply graveler/pgrefs_test.go:343]: 10x

File: graveler/pgrefs_test.go [hunk 4]
```diff
@@ -250,6 +252,139 @@ func TestPGRefManager_ListBranches(t *testing.T) {
...
+		if !errors.Is(err, graveler.ErrNotFound) {
+			t.Fatalf("expected ErrNotFound, got error: %v", err)
+		}
+		if commitID != nil {
+			t.Fatalf("get not existing commitID: %s, expected nil", *commitID)
+		}
+	})
+}
+
+func TestPGRefManager_SetTag(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (nopcoder) [reply graveler/pgrefs_test.go:290]: copy paste bug :(

File: graveler/pgrefs.go [hunk 5]
```diff
@@ -187,13 +187,9 @@ func (m *PGRefManager) GetTag(ctx context.Context, repositoryID RepositoryID, ta
...
 
-func (m *PGRefManager) SetTag(ctx context.Context, repositoryID RepositoryID, tagID TagID, commitID CommitID) error {
+func (m *PGRefManager) CreateTag(ctx context.Context, repositoryID RepositoryID, tagID TagID, commitID CommitID) error {
 	_, err := m.db.Transact(func(tx db.Tx) (interface{}, error) {
-		_, err := tx.Exec(`
-			INSERT INTO graveler_tags (repository_id, id, commit_id)
-			VALUES ($1, $2, $3)
-				ON CONFLICT (repository_id, id)
-				DO UPDATE SET commit_id = $3`,
+		_, err := tx.Exec(`INSERT INTO graveler_tags (repository_id, id, commit_id) VALUES ($1, $2, $3)`,
```
Comment (ozkatz) [inline graveler/pgrefs.go:192]: I suggest handling the SQL error and returning an `ErrTagAlreadyExists` error

**Review by ozkatz — APPROVED** [review]
Nice!!

**Pull request merged**